[PP] Forward only schedule

H-Huang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132177

`python test/distributed/pipelining/test_schedule_multiproc.py -k test_forward_only`

cc @XilunWu @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

H-Huang committed:
[draft] [PP] Forward only schedule

[ghstack-poisoned]

H-Huang committed:
Update on "[draft] [PP] Forward only schedule"


python test/distributed/pipelining/test_schedule_multiproc.py -k test_forward_only

cc XilunWu awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

lessw2020 commented on test/distributed/pipelining/test_schedule_multiproc.py:
```diff
@@ -56,6 +57,47 @@ def setUpClass(cls):
+
+        # Run
+        num_iters = 20
+        for _ in range(num_iters):
```
generic q - is there an easy way to verify that things looped successfully (i.e. seed value for rand and then confirm expected final values after 20 iters)?
right now the test confirms it ran w/o error, but maybe future test should also verify an expected output?

H-Huang replied:
Thanks! Added a validation check at the end of the test to compare the pipelined output with a reference model.

lessw2020 reviewed (approved):
looks great, thanks a ton for adding! 
- minor q on future test addition
- linting errors are easy fixes

H-Huang committed:
Update on "[draft] [PP] Forward only schedule"


`python test/distributed/pipelining/test_schedule_multiproc.py -k test_forward_only`

cc XilunWu awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

H-Huang committed:
Update on "[PP] Forward only schedule"


`python test/distributed/pipelining/test_schedule_multiproc.py -k test_forward_only`

cc XilunWu awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

H-Huang commented:
@pytorchbot merge

H-Huang committed:
Update on "[PP] Forward only schedule"


`python test/distributed/pipelining/test_schedule_multiproc.py -k test_forward_only`

cc XilunWu awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

H-Huang commented:
@pytorchbot merge
